From ea45f2202ea8059147746902d7f30315cdcdf209 Mon Sep 17 00:00:00 2001
From: Enrique Moron Ayuso <emoronayuso@gmail.com>
Date: Fri, 20 Mar 2015 01:18:46 +0100
Subject: [PATCH] Example coding exercise: include ramdom num

Signed-off-by: Enrique Moron Ayuso <emoronayuso@gmail.com>
---
 coding_exercise.c | 20 +++++++++++---------
 1 file changed, 11 insertions(+), 9 deletions(-)

diff --git a/coding_exercise.c b/coding_exercise.c
index 90a858c..a77bce8 100644
--- a/coding_exercise.c
+++ b/coding_exercise.c
@@ -31,6 +31,7 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <time.h>
+#include <stdio.h>
 
 #define TAM 100
 
@@ -64,7 +65,7 @@ void peermgr_init(PeerManager *mgr, unsigned int maxpeers)
 	mgr->_maxpeers = maxpeers;
 	mgr->_num_peers = 0;
 }
-"
+
 /**
  * Free any resources allocated by @mgr.
  */
@@ -85,9 +86,9 @@ void peermgr_cleanup(PeerManager *mgr)
  */
 void peermgr_saw_peer(PeerManager *mgr, PeerId peer, time_t timestamp)
 {
-	
+
 	if (mgr->_num_peers < mgr->_maxpeers) {
-		mgr->_tabla_peers[_num_peers +1] = peer;
+		mgr->_tabla_peers[mgr->_num_peers +1] = peer;
 		mgr->_timestamp = timestamp;
 		mgr->_num_peers++;
 	} else {
@@ -105,12 +106,13 @@ void peermgr_saw_peer(PeerManager *mgr, PeerId peer, time_t timestamp)
 bool peermgr_pick_random_peer(PeerManager *mgr, PeerId *peer)
 {
     /* TODO use random(3) and assume a seed has already been set */
-	long int num = random(3);
+	//long int num = random(3);
+	//srandom(3);
+	//char *initstate(3, char *state, mgr->_num_peers);
+	srand(time(NULL));
+	int num_random = rand() % mgr->_num_peers;
 	
-	if (num < _num_peers) {
-		*peer = mgr->_tabla_peers[num];
-		return true;
-	} else
-		return false;
+	*peer = mgr->_tabla_peers[num_random];
+	return true;
 
 }
-- 
2.1.4

